### PR TITLE
plot classify output 

### PR DESCRIPTION
**Pull request by kiepczi**
This PR adds plotting functionality to `pyani-classify`. Previously, it only generated a `.tsv` file with clique and singleton information, but now it also produces a plot with three stacked subplots:
1. Number of genomes in cliques vs. singletons at different %identity values.
2. Percentage of genomes in cliques and singletons across %identity.
3. Lifespan of each clique (range of identity where they exists).

The plot size scales based on the number of genomes—more genomes = bigger plot and po plot will be generated if:
- The comparison only has one genome.
- The initial graph is all singletons.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update

## Action Checklist

- [X] Work on a single issue/concept (if there are multiple separate issues to address, please use a separate pull request for each)
- [ ] Fork the `pyani-plus` repository under your own account (please [allow write access for repository maintainers](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
- [X] Set up an appropriate development environment (please see `CONTRIBUTING.md` for this repository)
- [X] Create a new branch with a short, descriptive name
- [X] Work on this branch
  - [X] style guidelines have been followed (please see `CONTRIBUTING.md` for this repository)
  - [X] new code is commented, especially in hard-to-understand areas
  - [ ] corresponding changes to documentation have been made
  - [X] tests for the change have been added that demonstrate the fix or feature works
- [X] Test locally with `pytest-plus -v` **non-passing code will not be merged**
- [ ] Rebase against `origin/master`
- [X] Check changes with linting/formatting tools before submission (please see `CONTRIBUTING.md` for this repository)
- [X] Commit branch
- [X] Submit pull request, describing the content and intent of the pull request
- [X] Request a code review
- [X] Continue the discussion at [`Pull requests` section](https://github.com/widdowquinn/pyani-plus/pulls) in the `pyan-plus` repository

### CODE REVIEW

File: pyani_plus/public_cli.py [hunk 1]
```diff
@@ -1109,10 +1112,29 @@ def cli_classify(  # noqa: C901, PLR0913
...
+    clique_data, clique_df = classify.compute_classify_output(
+        unique_cliques, method, outdir, column_map
+    )
     print(f"Wrote classify output to {outdir}")
+
+    # Only plot classify if more than one genome in comparisons and the initial graph consist of at least one clique
+    if not single_genome_run:
+        if set(clique_df["n_nodes"]) == {1}:
+            msg = "WARNING: All genomes are singletons. No plot can be generated."
+            sys.stderr.write(msg)  # pragma: no cover
```
Comment (peterjc) [inline pyani_plus/public_cli.py:1132]: This looks like an easy corner case to test - see `capsys` in other examples.

Comment (peterjc) [reply pyani_plus/public_cli.py:1132]: Actually, you do have a test for this - so just need to take off the "pragma: no cover" comment?

Comment (kiepczi) [reply pyani_plus/public_cli.py:1132]: I think it was tested in `test_classify_warnings`, or am I missing something?

File: pyani_plus/classify.py [hunk 2]
```diff
@@ -201,17 +206,201 @@ def get_unique_cliques(
...
+    ax3.yaxis.tick_right()
+    ax3.set_xlim(math.floor(dataframe[f"min_{score}"].min() * 100) / 100 - 0.01, 1.0)
+    ax3.set_ylim(-1, num_genomes)
+    ax3.axvline(x=0.95, color="red", linewidth=2, linestyle="--")
+    ax3.grid(True, linestyle="--", linewidth=0.5, alpha=0.7)  # noqa: FBT003
+
+    # Save the figure as a PDF file
+    plt.savefig(
+        outdir / f"{method}_classify_plot.pdf", format="pdf", bbox_inches="tight"
+    )
```
Comment (peterjc) [inline pyani_plus/classify.py:397]: Can you import the standard list of graphics formats, and loop over them to make PNG etc as well as PDF?

**Comment by widdowquinn** [comment]
This looks fantastic, but we should handle #298, #299, #300 before merging.

**Comment by widdowquinn** [comment]
(and if we can get the layout a bit tighter first, so much the better!)

**Comment by kiepczi** [comment]
https://github.com/pyani-plus/pyani-plus/pull/294/commits/11f4d77387eea2f7b049bd9dda08b040911c5e0b fixes #298 and fixes #300. 

Using `-tANI` instead of `tANI` values fixed the issue. In this commit, I also:
- Fixed the x-axis scale—it was set to 1.0 by default, but that doesn’t work for `tANI`.
- Set the vertical line to 0.95 for identity and -0.323 for tANI.

Next up, I’ll make this optional so people can use their own threshold if they want.
